src/vcf.rs

```diff
@@ -545,5 +604,730 @@ fn snp_dist(reference: &String, alternate: &String) -> i64 {
+        assert_eq!(snp_dist("ACGTACGT", "ACGTACGT"), 0);
+        assert_eq!(snp_dist("ACGT", "AAGT"), 1);
+        assert_eq!(snp_dist("ACGT", "AAAT"), 2);
+        assert_eq!(snp_dist("ACGT", "AAAA"), 3);
+        assert_eq!(snp_dist("ACGT", "NNNN"), 0);
+        assert_eq!(snp_dist("ACGT", "NAGT"), 1);
+
+        // Should panic
+        assert_panics!(snp_dist("ACGT", "ACG"));
+        assert_panics!(snp_dist("", "ACGT"));
+        assert_panics!(snp_dist("ACGT", ""));
+    }
+
+    #[test]
+    fn test_simplify_call() {
+        // Some trivial cases
+        assert_eq!(
+            VCFFile::simplify_call("ACGT".to_string(), "ACGT".to_string()),
+            vec![]
+        );
+        assert_eq!(
+            VCFFile::simplify_call("AAAAA".to_string(), "AAAAA".to_string()),
+            vec![]
+        );
+        assert_eq!(
+            VCFFile::simplify_call("ACGTACGT".to_string(), "ACGTACGT".to_string()),
+            vec![]
+        );
+        assert_eq!(
+            VCFFile::simplify_call("ACGT".to_string(), "AAGT".to_string()),
+            vec![(1, AltType::SNP, "A".to_string())]
+        );
+        assert_eq!(
+            VCFFile::simplify_call("ACGT".to_string(), "AAAT".to_string()),
+            vec![
+                (1, AltType::SNP, "A".to_string()),
+                (2, AltType::SNP, "A".to_string()),
+            ]
+        );
+        assert_eq!(
+            VCFFile::simplify_call("ACGT".to_string(), "AAAA".to_string()),
+            vec![
+                (1, AltType::SNP, "A".to_string()),
+                (2, AltType::SNP, "A".to_string()),
+                (3, AltType::SNP, "A".to_string()),
+            ]
+        );
+        // Some more complex cases
+        assert_eq!(
+            VCFFile::simplify_call("ACGT".to_string(), "ACG".to_string()),
+            vec![(3, AltType::DEL, "T".to_string())]
+        );
+        assert_eq!(
+            VCFFile::simplify_call("ACG".to_string(), "ACGT".to_string()),
+            vec![(2, AltType::INS, "T".to_string())]
+        );
+        assert_eq!(
+            VCFFile::simplify_call("ACGT".to_string(), "AGT".to_string()),
+            vec![(1, AltType::DEL, "C".to_string()),]
+        );
+
+        // Somewhat pathological cases
+        assert_eq!(
+            VCFFile::simplify_call("AAAAA".to_string(), "CCC".to_string()),
+            vec![
+                (3, AltType::DEL, "AA".to_string()),
+                (0, AltType::SNP, "C".to_string()),
+                (1, AltType::SNP, "C".to_string()),
+                (2, AltType::SNP, "C".to_string()),
+            ]
+        );
+
+        assert_eq!(
+            VCFFile::simplify_call("CCC".to_string(), "AAAAA".to_string()),
+            vec![
+                (2, AltType::INS, "AA".to_string()),
+                (0, AltType::SNP, "A".to_string()),
+                (1, AltType::SNP, "A".to_string()),
+                (2, AltType::SNP, "A".to_string()),
+            ]
+        );
```

Comment:
I'd probably want to add a test case like ACGT -> ACGAGT or something where the ins/del is in the middle

Reply:
though the indel is actually put at the end. So it's 3 snps and then an insertion

Reply:
Ah true, added on my other branch as I forgot I wasn't on this one